app/code/community/Bolt/Boltpay/Model/BoltOrder.php

```diff
@@ -365,17 +365,132 @@ protected function getCorrectedTotal($projectedTotal, $magentoDerivedCartData)
+     * remedied, the tail end of that address line is truncated to satisfy the restraint
+     *
+     * @param array $streetAddressArray   The street address data to be adjusted
+     * @param int $startingIndex
+     *
+     * @throws Mage_Core_Exception  thrown if there is no natural place break up the address,
+     */
+    private function reallocateAddressLines(&$streetAddressArray, $startingIndex=0){
+
+        /** @var Bolt_Boltpay_Helper_Bugsnag $bugsnag */
+        $bugsnag = Mage::helper('boltpay/bugsnag');
+
+        $startingIndex = min($startingIndex, 3);
+
+        ///////////////////////////////////////////
+        /// Make sure we have an array of length 4
+        ///////////////////////////////////////////
+        while (count($streetAddressArray) < 4) $streetAddressArray[] = '';
+        ///////////////////////////////////////////
+
+        $tempLine = $streetAddressArray[$startingIndex] = preg_replace('/\h+/', ' ', $streetAddressArray[$startingIndex]);
+
+        if ( strlen($tempLine) > 50 ) {
+            // The length of this line is too long.  We must try to correct it
+
+            if ( $startingIndex >= 3) {
+                // Since this is address line 4, the last available line, we will truncate the string
+                // to meet requirements and send a warning to Bugsnag
+                $streetAddressArray[$startingIndex] = substr($tempLine, 0, 50);
+
+                $message = "Cropped address line '$tempLine' to '".$streetAddressArray[$startingIndex]."'\nOriginal Address: ".var_export($streetAddressArray, true);
+                $bugsnag->notifyException(new Exception($message), array(), 'warning');
+
+                return;
+            }
+
+            ///////////////////////////////////////////////////////////
+            /// Try to find a natural splitting point in address line,
+            /// then split the string
+            ///////////////////////////////////////////////////////////
+            $breakPoint = strrpos(substr($tempLine, 0, 50), ' ');
+
+            if ( $breakPoint === false ) {
+                $message = "No natural break was found in address line '$tempLine'"."\nOriginal Address: ".var_export($streetAddressArray, true);
+                $bugsnag->notifyException(new Exception($message));
+                Mage::throwException($message);
+            }
+
+            $correctedAddressLine = substr($tempLine, 0, $breakPoint);
+            $extraAddressInfo = substr($tempLine, $breakPoint + 1);
+            ///////////////////////////////////////////////////////////
+
+            $streetAddressArray[$startingIndex] = $correctedAddressLine;
+
+            if ($this->hasEmptyAddressLine($streetAddressArray, $startingIndex)) {
+
+                // Since there is an empty space, we fill it by shifting
+
+                for ($i = $startingIndex + 1; $i < count($streetAddressArray); $i++) {
```

Comment:
We're going to get an OOB error here I think if we start at higher indices. Can we do a `foreach` on `array_slice($streetAddressArray, $startingIndex + 1)` instead?